src/Compilers/Core/Portable/Compilation/CompilationOptions.cs

```diff
@@ -226,6 +227,11 @@ public abstract class CompilationOptions
         public StrongNameProvider StrongNameProvider { get; protected set; }
 
         /// <summary>
+        /// Provides options for analyzer consumption that apply to specific syntax trees. Null if no syntax tree specific options exist.
```

Comment:
:question: Why can't we make this return a non-null default provider that just returns the compilation's `OptionSet` for all requests?